pandas/core/indexes/multi.py

```diff
@@ -1366,6 +1366,11 @@ def remove_unused_levels(self):
 
         changed = False
         for lev, lab in zip(self.levels, self.labels):
+            null_mask = lab == -1
+            if null_mask.any():
+                lab = lab[~null_mask]
```

Comment:
This does not work well.
```
df = pd.MultiIndex(levels=[['a', 'd', 'b'], ['w', 'x', 'y', 'z']], labels=[[0, 2, -1, 1, 1], [0, 0, 0, 0, 0]])
df.remove_unused_levels()
```

Reply:
Please refer this.
https://github.com/pandas-dev/pandas/blob/master/pandas/core/indexes/multi.py#L600